File: Lab1/src/vigenere/Task2.java
package vigenere;

import caesar.Caesar;

public class Task2 {

    public static void main(String[] args) {
        String in = "G0IFOFVMLRAPI1QJbEQDbFEYOFEPJxAfI10JbEMFIUAAKRAfOVIfOFkYOUQFI15ML1kcJFUeYhA4IxAeKVQZL1VMOFgJbFMDIUAAKUgFOElMI1ZMOFgFPxADIlVMO1VMO1kAIBAZP1VMI14ANRAZPEAJPlMNP1VMIFUYOFUePxxMP19MOFgJbFsJNUMcLVMJbFkfbF8CIElMfgZNbGQDbFcJOBAYJFkfbF8CKRAeJVcEOBANOUQDIVEYJVMNIFwVbEkDORAbJVwAbEAeI1INLlwVbF4JKVRMOF9MOUMJbEMDIVVMP18eOBADKhALKV4JOFkPbFEAK18eJUQEIRBEO1gFL1hMO18eJ1UIbEQEKRAOKUMYbFwNP0RMNVUNPhlAbEMFIUUALUQJKBANIl4JLVwFIldMI0JMK0INKFkJIkRMKFUfL1UCOB5MH1UeJV8ZP1wVYBAbPlkYKRAFOBAeJVcEOBACI0dAbEkDORAbJVwAbF4JKVRMJURMOF9MKFUPJUAEKUJMOFgJbF4JNERMI14JbFEfbEcJIFxCbHIJLUJMJV5MIVkCKBxMOFgJPlWOzKkfbF4DbEMcLVMJPx5MRlgYOEAfdh9DKF8PPx4LI18LIFVCL18BY1QDL0UBKV4YY1RDfXg1e3QAYQUFOGkof3MzK1sZKXIaOnIqPGRYD1UPC2AFHgNcDkMtHlw4PGFDKVQFOA8ZP0BRP1gNPlkCKw==";

        int keyLength = Vigenere.analyzeKeyLength(in);
        System.out.println("[KEY LENGTH]: " + keyLength);

        String[] blocks = Vigenere.divideIntoBlocks(in, keyLength);
        for (int i = 0; i < blocks.length; i++) {
            blocks[i] = Caesar.decodeXorBruteforce(blocks[i]);
            System.out.println(blocks[i]);
        }
        System.out.println("[RESULT]: " + Vigenere.margeBlocks(blocks, keyLength));

//        System.out.println(Vigenere.decodeXor(in, new String(new char[]{'*', '*', '*', '*'})));
    }
}

File: Lab1/Lab1/Task2/Task2.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab1
{
    class Task2
    {
        public static void Run(string input)
        {
            string inputBase64Decoded = ASCIIEncoding.ASCII.GetString(Convert.FromBase64String(input));
            //Console.WriteLine(inputBase64Decoded);

            Vigenere.AnalyzeKeyLength(inputBase64Decoded);
            int keyLength = 3;

            Result result = Vigenere.DecodeXorBruteforce(inputBase64Decoded, keyLength);
            Console.WriteLine(result);

            /*
             * Write a code to attack some simple substitution cipher. To reduce the complexity of this one we will use only uppercase letters, so the keyspace is only 26! 
             * To get this one right automatically you will probably need to use some sort of genetic algorithm (which worked the best last year), 
             * simulated annealing or gradient descent. Seriously, write it right now, you will need it to decipher the next one as well. 
             * Bear in mind, thereSs☼s no spaces.
             * https://docs.google.com/document/d/1HY7Dl-5itYD3C_gkueBvvBFpT4CecGPiR30BsARlTpQ/edit?usp=sharing+.
             */
        }
    }
}
